src/components/CartProvider.jsx

```diff
@@ -0,0 +1,34 @@
+import { createContext, useState } from "react";
+
+export const CartContext = createContext();
+
+export default function CartProvider({ children }) {
+    const key = "cart";
+
+    const [cart, setCart] = useState(console.log(localStorage[key]) || [])
+
+    function addToCart(product) {
+        setCart([...cart, product])
+        localStorage[key] = JSON.stringify(cart)
```

Comment:
```suggestion
        localStorage[key] = JSON.stringify([...cart, product])
```

At the moment it looks like you are saving the old cart contents